### PR TITLE
gh-119771: adjust errno on overflows in _Py_c_pow()

### PR DESCRIPTION
**Pull request by skirpichev**
Before we did this in complex_pow() and behaviour of the public C API function ``_Py_c_pow()`` was different from the pure-python pow().

* Issue: gh-119771

----
📚 Documentation preview 📚: https://cpython-previews--120256.org.readthedocs.build/

### CODE REVIEW

File: Lib/test/test_capi/test_complex.py [hunk 1]
```diff
@@ -226,7 +226,10 @@ def test_py_c_pow(self):
 
         self.assertEqual(_py_c_pow(0j, -1)[1], errno.EDOM)
         self.assertEqual(_py_c_pow(0j, 1j)[1], errno.EDOM)
-        self.assertEqual(_py_c_pow(*[DBL_MAX+1j]*2)[0], complex(*[INF]*2))
+        self.assertEqual(_py_c_pow(*[DBL_MAX+1j]*2),
+                         (complex(*[INF]*2), errno.ERANGE))
+        self.assertEqual(_py_c_pow(DBL_MAX+1j, 2),
+                         (complex(*[INF]*2), errno.ERANGE))
```
Comment (vstinner) [inline Lib/test/test_capi/test_complex.py]: The syntax is a little bit surprising. I suggest something more regular:

```suggestion
        max_num = DBL_MAX+1j
        self.assertEqual(_py_c_pow(max_num, max_num),
                         (complex(INF, INF), errno.ERANGE))
        self.assertEqual(_py_c_pow(max_num, 2),
                         (complex(INF, INF), errno.ERANGE))
```

File: Objects/complexobject.c [hunk 2]
```diff
@@ -546,12 +548,12 @@ complex_pow(PyObject *v, PyObject *w, PyObject *z)
     // a faster and more accurate algorithm.
     if (b.imag == 0.0 && b.real == floor(b.real) && fabs(b.real) <= 100.0) {
         p = c_powi(a, (long)b.real);
+        _Py_ADJUST_ERANGE2(p.real, p.imag);
```
Comment (skirpichev) [inline Objects/complexobject.c:572]: That might be moved to ``c_powi()``, but that will make patch slightly larger.  I think this is fine, as ``c_powi()`` is not a part of C API.

**Review by vstinner — APPROVED** [review]
LGTM

**Comment by skirpichev** [comment]
@vstinner, do you want a second opinion on this?

**Review by vstinner — COMMENTED** [review]
LGTM

**Comment by vstinner** [comment]
Merged, thanks. I made minor edits on the commit message.

**Pull request merged**